tests/python/gaia-ui-tests/gaiatest/apps/homescreen/app.py

```diff
@@ -172,14 +110,45 @@ def number_of_columns(self):
         Wait(self.marionette).until(lambda m: element.get_attribute('cols') is not None)
         return int(element.get_attribute('cols'))
 
-    class InstalledApp(PageRegion):
+    def delete_app(self, app_manifest):
+        app = self.installed_app(app_manifest)
+        remove_action_element = self.marionette.find_element(*self._remove_locator)
+        app.move_to(remove_action_element)
+
+        from gaiatest.apps.homescreen.regions.confirm_dialog import ConfirmDialog
+        return ConfirmDialog(self.marionette)
+
+    class HomescreenElement(PageRegion):
 
         @property
         def manifest_url(self):
             return self.root_element.get_attribute('data-identifier')
 
+        @property
+        def is_app_installed(self):
+            self.marionette.switch_to_shadow_root(self.root_element)
```

Comment:
Just wondering, when we switch to shadow root, it is assumed that at the end of the method you already switch back, as you do in here right?  we should add this in our list of  best practices as well.